### PR TITLE
feat: support contributors

### PR DESCRIPTION
**Pull request by jonathanCaamano**
Closes https://github.com/inditex/iac-sentijobs/issues/142

### CODE REVIEW

File: workflow/filter/filter.go [hunk 1]
```diff
@@ -36,18 +36,21 @@ func CreateListOptions(ctx context.Context, listOptions *metav1.ListOptions) *me
...
+		return false
+	}
+	for key, relationship := range ctx.Value(auth.ClaimsKey).(*argoTypes.Claims).TeamFilterClaims.Values {
+		if key == labels[ctx.Value(auth.ClaimsKey).(*argoTypes.Claims).TeamFilterClaims.Label] {
+			if ctx.Value(auth.ClaimsKey).(*argoTypes.Claims).TeamFilterClaims.EnvToFilter != "pro" {
+				return true
+			}
+			if relationship == "Owner" && ctx.Value(auth.ClaimsKey).(*argoTypes.Claims).TeamFilterClaims.Group == "writer" {
+				return true
 			}
```
Comment (Dav1dPF) [inline workflow/filter/filter.go:52]:
```suggestion
	for key, relationship := range ctx.Value(auth.ClaimsKey).(*argoTypes.Claims).TeamFilterClaims.Values {
		if key != labels[ctx.Value(auth.ClaimsKey).(*argoTypes.Claims).TeamFilterClaims.Label] {
		      continue
		}
                if ctx.Value(auth.ClaimsKey).(*argoTypes.Claims).TeamFilterClaims.EnvToFilter != "pro" || ( relationship == "Owner" && ctx.Value(auth.ClaimsKey).(*argoTypes.Claims).TeamFilterClaims.Group == "writer"){
                return true
                }
```

**Review by Dav1dPF — CHANGES_REQUESTED** [review]

File: server/auth/devhub/aplication.go [hunk 2]
```diff
@@ -74,12 +74,15 @@ func GetRolesAndServices(result *ApiStruct, services, roles []string) ([]string,
...
-			if !slices.Contains(services, project.Key) {
-				services = append(services, project.Key)
+			_, ok := services[project.Key]
+			if ok {
+				if services[project.Key] != "Owner" && project.RelationshipType == "Owner" {
+					services[project.Key] = project.RelationshipType
+				}
+			} else {
+				services[project.Key] = project.RelationshipType
 			}
```
[the remark this answers is not in the document]
Comment (jonathanCaamano) [reply server/auth/devhub/aplication.go:84]: Done

File: server/cronworkflow/cron_workflow_server.go [hunk 3]
```diff
@@ -54,6 +55,7 @@ func (c *cronWorkflowServiceServer) ListCronWorkflows(ctx context.Context, req *
 }
 
 func (c *cronWorkflowServiceServer) CreateCronWorkflow(ctx context.Context, req *cronworkflowpkg.CreateCronWorkflowRequest) (*v1alpha1.CronWorkflow, error) {
+	var hasPermission bool
```
[the remark this answers is not in the document]
Comment (jonathanCaamano) [reply server/cronworkflow/cron_workflow_server.go:58]: Done

File: server/cronworkflow/cron_workflow_server.go [hunk 4]
```diff
@@ -66,6 +68,10 @@ func (c *cronWorkflowServiceServer) CreateCronWorkflow(ctx context.Context, req
 	if err != nil {
 		return nil, sutils.ToStatusError(err, codes.InvalidArgument)
 	}
+	hasPermission = filter.ForbidActionsIfNeeded(ctx, req.CronWorkflow.Labels)
+	if !hasPermission {
+		return nil, status.Error(codes.PermissionDenied, "permission denied")
+	}
```
[the remark this answers is not in the document]
Comment (jonathanCaamano) [reply server/cronworkflow/cron_workflow_server.go:74]: Done

File: server/workflow/workflow_server.go [hunk 5]
```diff
@@ -682,8 +681,8 @@ func (s *workflowServer) WorkflowLogs(req *workflowpkg.WorkflowLogRequest, ws wo
 	return sutils.ToStatusError(s.PodLogs(req, ws), codes.Internal)
 }
 
-func (s *workflowServer) getWorkflow(ctx context.Context, wfClient versioned.Interface, namespace string, name string, options metav1.GetOptions) (*wfv1.Workflow, error) {
-	var hasService bool
+func (s *workflowServer) getWorkflow(ctx context.Context, wfClient versioned.Interface, namespace string, name string, options metav1.GetOptions, isPodLogs bool) (*wfv1.Workflow, error) {
+	var hasPermission bool
```
[the remark this answers is not in the document]
Comment (jonathanCaamano) [reply server/workflow/workflow_server.go:685]: Done

File: workflow/filter/filter.go (hunk 1 above)
Comment (jonathanCaamano) [reply workflow/filter/filter.go:52]: done

File: workflow/filter/filter.go [hunk 6]
```diff
@@ -36,18 +36,20 @@ func CreateListOptions(ctx context.Context, listOptions *metav1.ListOptions) *me
...
+		return true
+	}
+	if len(ctx.Value(auth.ClaimsKey).(*argoTypes.Claims).TeamFilterClaims.Values) <= 0 {
+		return false
+	}
+	for key, relationship := range ctx.Value(auth.ClaimsKey).(*argoTypes.Claims).TeamFilterClaims.Values {
+		if key != labels[ctx.Value(auth.ClaimsKey).(*argoTypes.Claims).TeamFilterClaims.Label] {
+			continue
+		}
+		if ctx.Value(auth.ClaimsKey).(*argoTypes.Claims).TeamFilterClaims.EnvToFilter != "pro" || (relationship == "Owner" && ctx.Value(auth.ClaimsKey).(*argoTypes.Claims).TeamFilterClaims.Group == "writer") {
```
Comment (DanielDorado) [inline workflow/filter/filter.go:49]: Here, we do not want to know what environment we are in. Only if for a service it is a "writer" or a "reader".

This should be done with the SSO configuration.

- What Roles (and Relationship: Owner or Contributor) have what Group.

This way, we do not pollute Argo workflows with this logic.

**Review by DanielDorado — CHANGES_REQUESTED** [review]

File: server/auth/devhub/aplication.go [hunk 7]
```diff
@@ -7,7 +7,7 @@ import (
 )
 
 type GroupAndServices struct {
-	Services []string
+	Services map[string]string
```
[the remark this answers is not in the document]
Comment (jonathanCaamano) [reply server/auth/devhub/aplication.go]: Done

File: config/sso_extended.go [hunk 8]
```diff
@@ -39,8 +41,11 @@ func RbacDelegateToLabel(ctx context.Context, mail string, apiUrl, apiEndpoint,
 	}
```
[the remark this answers is not in the document]
Comment (jonathanCaamano) [reply config/sso_extended.go:40]: Done

File: workflow/filter/filter.go (hunk 6 above)
Comment (jonathanCaamano) [reply workflow/filter/filter.go]: Done

**Review by DanielDorado — APPROVED** [review]

**Pull request merged**